src/librustc_mir/transform/deaggregator.rs

```diff
@@ -55,6 +56,23 @@ impl MirPass for Deaggregator {
                         }
                         active_field_index
                     }
+                    AggregateKind::Generator(..) => {
```

Comment:
@eddyb Does the codegen backend assume there's no aggregates or is this just an optimization?

Reply:
It's not turned on by default AFAIK. We might change that in the future though, so it'd be good to have it work, for completeness.